### DISCUSSION TITLE
Multithreaded rendering attempt

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3824] [General]
I made a fleeting vain attempt at adding multithreaded rendering and discovered that, near as I can see, it isn't any faster or slower.  Perhaps searching for rectangles dominates?

Any thoughts? Anyone else who wants to have a go? :-)

https://github.com/The-OpenROAD-Project/OpenROAD/pull/3813

My theory, which didn't seem to hold up, was:

KISS multithreaded rendering: one subrectangle per thread.

This means that a search for objects to draw into each subrectangle is done once per thread, so that might be more wasteful than searching for rectangles once and then rendering one layer per thread.

However, in terms of the rendering time this might not matter, because no rendering starts until the search for objects to draw within a rectangle as completed.

**Comment by Iyury1** [comment]
I started looking at multithreading with one layer per thread. I can only work on this on weekends though, so I won't have a PR ready until the last week of August or first week of September if you can wait that long.

**Reply by oharboe** [reply]
It looks like your approach will be more successful because it will search for rectangles in parallel. I have closed my PR, but it is still available to look at and try out.

There's no rush for my part, I was just curious and gave it a go.
